dom/nodes/insertion-effects/Node-appendChild-script-and-style-from-fragment.tentative.html

```diff
@@ -0,0 +1,30 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>Node.appendChild: inserting a script and a style from a fragment</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<body>
+<script>
+const happened = []
+const style = document.createElement("style");
+let styleSheet = null;
+test(() => {
+  style.appendChild(new Text("body {}"));
+  const script = document.createElement("script");
+  script.textContent = `
+    happened.push(style.sheet);
+    style.appendChild(new Text("body {}"));
+    happened.push(style.sheet.cssRules.length);
+    styleSheet = style.sheet;
+`;
+
+  const df = document.createDocumentFragment();
+  df.appendChild(script);
+  df.appendChild(style);
+
+  assert_array_equals(happened, []);
+  document.body.appendChild(df);
+  assert_array_equals(happened, [null, 2]);
```

Comment:
But "becomes connected" is part of the insertion steps, so how come (2) is before (3) ? Don't we run all the insertion steps in order?

Reply:
Yeah so today, the "becomes connected" insertion steps are run immediately per-node during DOM insertion. In our model, we will keep the "becomes connected" insertion steps running immediately per-node during DOM insertion, and simply lift out any script-executing steps into the post-insertion steps (or whatever we call it). That means for `<style>` elements, nothing changes (so (2) happens in the same timing as it would today). But for script elements (whose current insertion steps involve executing script), we change that to happen *after* all DOM mutations happen (i.e., after all "normal" insertion steps run) and script executes in the special post-insertion steps.

Reply:
I think this can be resolved given our discussion offline/elsewhere.